src/JavascriptVM/JavascriptVM.tsx

```diff
@@ -50,6 +52,7 @@ export const VMProvider: FunctionComponent = ({ children }) => {
   );
   const dispatch = useDispatch<AppDispatch>();
   const code = useSelector(getCode);
+  const resetLink = useRef<() => void | undefined>();
```

Comment:
I think this is a good reason for us to hoist the Sim3D instance up to it's own provider or into one that provides both. There's a lot of Refs and I'm a little afraid they'll make debugging harder in the future when we have more than one view as things will get unmounted and destroyed

Reply:
are you suggesting we do this now? before landing this pr?

Reply:
Yeah, maybe this is a good time for a refactor. Since the VM and the Simulator will need to talk to each other maybe what we can do is have one SimulatorProvider that holds both the interpreter insurance and the Sim3D instance. Then it'd be trivial to connect the two together and for the rest of the application to control them. 

The question is then do we keep redux or simply make the Simulator provider the source of truth for the application?